src/app/onboarding/components/onboarding/onboarding.component.ts

```diff
@@ -43,6 +67,9 @@ export class OnboardingComponent implements OnInit {
     if (correspondingRoute) {
       this.location.go('app/onboarding/' + correspondingRoute.path);
     }
+    if (screen === OnboardingScreen.done) {
+      this.router.navigate(['/app']);
```

Comment:
What is the difference between `location.go` and `router.navigate`?

Reply:
I believe `location.go` is some kind of `history.pushState` type thing where as `router.navigate` is a more "hard" navigation... if that makes sense?

Reply:
Ah, I'm afraid it does not. :S I might be missing some background!

When would you use one or the other? Or, how concerned should I be that we're using multiple different types of navigation?

I don't think this is a blocker for deploying this, but I would like to understand!

Reply:
`location.go` sets the URL in the browser window without actually refreshing the page, so we can keep the current component state intact while we go through the different onboarding screens. `router.navigate` actually causes a real browser navigation event to happen, which causes the page to reload. I believe this is the difference between the two calls.

Reply:
Aha! That is excellent context, thank you.

So here, after we finish onboarding, we want the page to reload so that the app asks for the default archive and populates all the state it needs as on first login?

Reply:
Yes! Also `location.go` doesn't actually cause Angular to reload any modules. Just as a sanity check I replaced `router.navigate` with `location.go` and while it set the URL correctly, I'm still stuck on the onboarding page.

Note the browser URL:
![2022-02-17-130423](https://user-images.githubusercontent.com/9145247/154562131-754a0d6a-bafa-41a3-8aad-57a5b02c2d60.png)

Reply:
Awesome - thanks again for explaining!